Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemScriptingHandler.cpp

```diff
@@ -20,6 +20,44 @@
 
 namespace AzToolsFramework::Prefab
 {
+    static AZStd::unique_ptr<AZ::Entity> CreateContainerEntityAndParentEntities(
+        const AZStd::vector<AZ::Entity*> entities, const AZStd::string& filePath)
+    {
+        bool result = false;
+        [[maybe_unused]] AZ::EntityId commonRoot;
+        EntityList topLevelEntities;
+        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+            result, &AzToolsFramework::ToolsApplicationRequestBus::Events::FindCommonRootInactive, entities, commonRoot, &topLevelEntities);
+
+        auto containerEntity = AZStd::make_unique<AZ::Entity>();
+
+        containerEntity->CreateComponent<Components::EditorLockComponent>();
+        containerEntity->CreateComponent<Components::EditorVisibilityComponent>();
+        containerEntity->CreateComponent<Prefab::EditorPrefabComponent>();
+
+        {
+            auto transformComponent = containerEntity->CreateComponent<Components::TransformComponent>();
+            // Because procedural prefabs need to be deterministic we need to set the component ID to something unique and non-random
+            // The prefab that references the proc prefab will store a patch that references the transform component by it's component ID
+            // If this ID is not stable, the proc prefab will lose its position, parenting, etc data next time it is regenerated
+            auto hash = TypeHash64(reinterpret_cast<const uint8_t*>(filePath.data()), filePath.length(), AZ::HashValue64{ 0 });
```

Comment:
is filepath a relative path?  It also needs to be stable between different peoples machines, too, ideally.  So people can collaborate and pass the files around.

Reply:
Yes, it's a relative path.

However, I don't think this is needed anymore. I believe this was needed previously component aliases were being set as 'Component_[%id]'. And this hashing scheme was being used make sure that the alias always remains the same. However, since we switched to using typename based aliases, the alias of the transform component now should always be "{27F1E1A1-8D9D-4C3B-BD3A-AFB9762449C0} TransformComponent"

I'll remove this part and see if this still works.

Reply:
Thanks for the suggestions. But I've verified that this hashing is no longer necessary. So I've removed it. Persistent overrides are now guaranteed because we use typename based component aliases.